### ISSUE TITLE
Simulation or live preview of configured rule instances' behaviour

### ISSUE THREAD

**Issue by phager-at** [daledavies/moodle-tool_registrationrules#52]
First of all this is certainly out of scope for the initial release, but I wanted to write the idea down, before it's lost.

We could implement a simulation, how the configured set of rule instances work given certain input.
Especially if we convert the rule instance management to an reactive UI the simulation/preview could update it's results with configuration changes made above.

This could aid (new) administrators in configuring the rule instances correctly for their use cases.

- each rule instance type can provide simulation input fields, to let the user manually control the simulated registration's data
- the simulation can then use the user provided data to check each rule for it's result (pre- and post-data) and display the result and all related feedback and validation messages

Blocked by #51
